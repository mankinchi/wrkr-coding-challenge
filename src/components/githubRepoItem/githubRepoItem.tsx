import { format } from "date-fns";
import { ExternalLink, Star } from "react-feather";
import { GithubRepo } from "../../typings/githubRepo";

interface Props {
	repo: GithubRepo;
}

export const GithubRepoItem = (props: Props) => {
	const {
		repo: {
			name,
			description,
			licenseName,
			stars,
			topics,
			lastUpdate,
			language,
			link,
		},
	} = props;

	const [main, ...rest] = name.split("/").reverse();

	return (
		<div className="flex gap-1">
			<div className="shrink-0 pt-2">
				<img
					src="/logo/repo-16.svg"
					alt="repo"
				/>
			</div>
			<div className="flex flex-col gap-2">
				<div className="text-lg text-blue-600">
					<a
						href={link}
						target="_blank"
						rel="noreferrer"
						className="hover:underline"
					>
						<span>{rest.join("/")}/</span>
						<span className="font-bold">{main}</span>
						<ExternalLink
							size={14}
							className="ml-1 inline align-super"
						/>
					</a>
				</div>
				{description && (
					<div className="select-none">{description}</div>
				)}
				{topics && topics.length !== 0 && (
					<div className="flex select-none flex-wrap gap-2">
						{topics.map((topic) => (
							<div
								key={topic}
								className="w-fit whitespace-nowrap rounded-lg bg-cyan-200 px-2 py-0.5 text-sm"
							>
								{topic}
							</div>
						))}
					</div>
				)}
				<div className="flex select-none items-center gap-5 text-sm text-slate-400">
					<div className="flex items-center gap-1">
						<Star size={12} />
						<span>{stars}</span>
					</div>
					{language && <div>{language}</div>}
					{licenseName && <div>{licenseName}</div>}
					<div>Updated {format(lastUpdate, "dd/MM/yyyy")}</div>
				</div>
			</div>
		</div>
	);
};
