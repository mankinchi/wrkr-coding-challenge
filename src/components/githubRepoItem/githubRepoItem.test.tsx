import { render } from "@testing-library/react";
import { GithubRepoItem } from "./githubRepoItem";

describe("Github Repo Item component", () => {
	it("should render correctly", () => {
		const { container } = render(
			<GithubRepoItem
				repo={{
					id: 1,
					name: "Test Repo",
					description: "Random description",
					licenseName: "MIT",
					language: "Typescript",
					stars: 10000,
					topics: ["UI", "Test", "123"],
					lastUpdate: new Date(2022, 18, 11, 9),
					link: "https://google.com",
				}}
			/>,
		);
		expect(container).toMatchSnapshot();
	});

	it("should render with optional attributes correctly", () => {
		const { container } = render(
			<GithubRepoItem
				repo={{
					id: 1,
					name: "Test Repo",
					stars: 10000,
					lastUpdate: new Date(2022, 18, 11, 9),
					link: "https://google.com",
				}}
			/>,
		);
		expect(container).toMatchSnapshot();
	});
});
