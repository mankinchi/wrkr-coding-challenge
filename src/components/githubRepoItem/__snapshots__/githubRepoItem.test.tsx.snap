// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Github Repo Item component should render correctly 1`] = `
<div>
  <div
    class="flex gap-1"
  >
    <div
      class="shrink-0 pt-2"
    >
      <img
        alt="repo"
        src="/logo/repo-16.svg"
      />
    </div>
    <div
      class="flex flex-col gap-2"
    >
      <div
        class="text-lg text-blue-600"
      >
        <a
          class="hover:underline"
          href="https://google.com"
          rel="noreferrer"
          target="_blank"
        >
          <span>
            /
          </span>
          <span
            class="font-bold"
          >
            Test Repo
          </span>
          <svg
            class="ml-1 inline align-super"
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
            />
            <polyline
              points="15 3 21 3 21 9"
            />
            <line
              x1="10"
              x2="21"
              y1="14"
              y2="3"
            />
          </svg>
        </a>
      </div>
      <div
        class="select-none"
      >
        Random description
      </div>
      <div
        class="flex select-none flex-wrap gap-2"
      >
        <div
          class="w-fit whitespace-nowrap rounded-lg bg-cyan-200 px-2 py-0.5 text-sm"
        >
          UI
        </div>
        <div
          class="w-fit whitespace-nowrap rounded-lg bg-cyan-200 px-2 py-0.5 text-sm"
        >
          Test
        </div>
        <div
          class="w-fit whitespace-nowrap rounded-lg bg-cyan-200 px-2 py-0.5 text-sm"
        >
          123
        </div>
      </div>
      <div
        class="flex select-none items-center gap-5 text-sm text-slate-400"
      >
        <div
          class="flex items-center gap-1"
        >
          <svg
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
          <span>
            10000
          </span>
        </div>
        <div>
          Typescript
        </div>
        <div>
          MIT
        </div>
        <div>
          Updated 
          11/07/2023
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Github Repo Item component should render with optional attributes correctly 1`] = `
<div>
  <div
    class="flex gap-1"
  >
    <div
      class="shrink-0 pt-2"
    >
      <img
        alt="repo"
        src="/logo/repo-16.svg"
      />
    </div>
    <div
      class="flex flex-col gap-2"
    >
      <div
        class="text-lg text-blue-600"
      >
        <a
          class="hover:underline"
          href="https://google.com"
          rel="noreferrer"
          target="_blank"
        >
          <span>
            /
          </span>
          <span
            class="font-bold"
          >
            Test Repo
          </span>
          <svg
            class="ml-1 inline align-super"
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
            />
            <polyline
              points="15 3 21 3 21 9"
            />
            <line
              x1="10"
              x2="21"
              y1="14"
              y2="3"
            />
          </svg>
        </a>
      </div>
      <div
        class="flex select-none items-center gap-5 text-sm text-slate-400"
      >
        <div
          class="flex items-center gap-1"
        >
          <svg
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
          <span>
            10000
          </span>
        </div>
        <div>
          Updated 
          11/07/2023
        </div>
      </div>
    </div>
  </div>
</div>
`;
